src/observables/FromEventPatternObservable.ts

```diff
@@ -0,0 +1,72 @@
+import Scheduler from '../Scheduler';
+import Observable from '../Observable';
+import Subscription from '../Subscription';
+import tryCatch from '../util/tryCatch';
+import {errorObject} from '../util/errorObject';
+
+export default class FromEventPatternObservable<T, R> extends Observable<T> {
+
+  static create<T, R>(addHandler: (handler: Function) => void, removeHandler: (handler: Function) => void, selector?: (x: R) => T, scheduler?: Scheduler) {
+    return new FromEventPatternObservable(addHandler, removeHandler, selector, scheduler);;
```

Comment:
the addHandler should not return void, as it currently works, it returns whatever is returned invoking the addHandler.  This is useful in places like Dojo which return a handle for unsubscription instead of a subscribe/unsubscribe pair